Make fundraising banner a more permanent feature (Fixes #7185)

alexgibson opened the pull request:
## Description
- Add `page_banner` block to base template.
- Create` includes/banners/base.html`
- Move fundraising banner to `includes/banners/fundraiser.html`.
- Split out base CSS banner styles from fundraising banner.
- Refactor fundraising banner JS to `mozilla-banner.js`.
- Add unit tests for JS.

## Issue / Bugzilla link
#7185

## Testing
- [x] Fundraising banner should still look / function the same as [on dev](https://www-dev.allizom.org/en-US/).

alexgibson commented:
Note to reviewer: whilst working on this I uncovered a bug with the current fundraising form (see https://github.com/mozilla/bedrock/issues/8606). I'm not quite sure what to do about this just yet, but it hopefully shouldn't effect the refactor here.

craigcook commented on bedrock/mozorg/templates/mozorg/home/home.html:
```diff
@@ -36,6 +36,12 @@
 
+{% block page_banner %}
+  {% if switch('home-fundraising-sept2019') %}
+    {% include 'includes/banners/fundraiser.html' %}
```
On further thought, maybe it should be its own CSS bundle and added to pages the same way we add the JS?

alexgibson replied:
Oh I thought I did that, but you’re right I only did it for the JS 👍

alexgibson commented on bedrock/base/templates/base-pebbles.html:
```diff
@@ -94,6 +94,8 @@
       {% include 'includes/protocol/navigation/menu-mozilla/index.html' %}
     {% endblock %}
 
+    {% block page_banner %}{% endblock %}
```
Can you explain a bit about why you feel it should come above the navigation? This was actually the first thing I thought to do here, as I always felt like it was out of place above and made the nav underneath a potentially confusing UX. I feel making it appear below the nav fits in better with the hierarchy of the page, and more like a permanent feature (vs feeling like an injected ad?). I’d be interested in opinions from others here too.

craigcook replied:
Well to be honest I've always thought of this as an injected ad and not part of the page anyway. I was just noting that it's currently above the nav so moving it below is a change, but maybe it's a change for the better. We should see with @dzingeek and @justinkropp think.

alexgibson replied:
@dzingeek @justinkropp to help give a little more context to this conversation, we're talking about making our current system for banners more reusable as a feature on the site. Right now the only banner we have is the homepage fundraising banner, but in the future we could show other things, such as a promo for Fenix as an example.

### Currently, the banner appears above the main site navigation like so:

![image](https://user-images.githubusercontent.com/400117/75972727-d26dba80-5ecb-11ea-8fad-e91387cc6f93.png)

### With this change, I'm proposing we move it below the nav:

![image](https://user-images.githubusercontent.com/400117/75972809-f3cea680-5ecb-11ea-972f-fe56e97f866e.png)

My thought here is that banners will fits a bit better with the general page hierarchy of the site when they appear below the navigation, but I'm open to keeping it as-is if others disagree.

justinkropp replied:
@alexgibson Thanks for tackling this and for asking for feedback. I'm on @craigcook's side here in thinking that the banner makes more sense to me -- from an overall user experience perspective -- living above the primary navigation. From some preliminary research, similar fundraising banners typically live at the very top of the page above the nav. Placing the banner below the nav will, imo, give the banner primary importance and potentially confuse the user into thinking that it is permanent page content. 

I really like this project and would like to see the design of this banner evolve a bit and get themed: one theme for the normal mozilla fundraising banner, and another theme for more generic Firefox promotional messaging.

alexgibson replied:
Ok, I'll leave it as-is for now - although we should consider this is not just a fundraising banner anymore. It may evolve to have multiple uses. 

Personally I question whether the current positioning is somewhat confusing, as users new to the site might typically expect navigation to appear at the top of the page, and scrolling down can then trigger opening the menu panels unexpectedly.

dzingeek replied:
I also agree with @craigcook and @justinkropp however I do get @alexgibson concerns. I think this might be a good candidate for some user testing to see how people respond to it from different placements. Antoher great point from @alexgibson is that this is no longer just a fundraiser banner and could have multiple uses. So my concern there is how often will this be used? If it's on the page more than it's not than we shoudl re-evalute it's placement a bit more.

alexgibson replied:
> So my concern there is how often will this be used? If it's on the page more than it's not than we shoudl re-evalute it's placement a bit more.

Yeah, I think for now this is fine to leave as-is. We're not (yet) building any other types of banner. But it may be worth re-evaluation when we make further use of the feature.

Thanks for the feedback all!

alexgibson commented:
Thanks for the review @craigcook. I've added a commit with the following:

- Fundraising banner now has it's own CSS bundle.
- Updated switch name to just `fundraising-banner`.
- Added docs for creating banners and managing the fundraising banner.

Whilst using the fundraising banner on any page could be a possibility, I'm focusing this PR on making future banners easy to implement. The current fundraising banner's strings are tied to the home page, so it can't really be used elsewhere in it's current state anyway. Hopefully we can fix this somehow when we migrate over to Fluent.
